src/plugin/ComponentManager.h

```diff
@@ -119,24 +119,59 @@ class ComponentManager {
 
     /**
      * @param name The canonical name of a component.
+     * @param consider_remappings If true, component mappings will be honored
+     * if the original component is disabled.
      * @return The component associated with the name or a null pointer if no
      * such component exists.
      */
-    C* Lookup(const std::string& name) const;
+    C* Lookup(const std::string& name, bool consider_remappings = true) const;
 
     /**
      * @param name A component tag.
+     * @param consider_remappings If true, component mappings will be honored
+     * if the original component is disabled.
      * @return The component associated with the tag or a null pointer if no
      * such component exists.
      */
-    C* Lookup(const zeek::Tag& tag) const;
+    C* Lookup(const zeek::Tag& tag, bool consider_remappings = true) const;
 
     /**
      * @param name A component's enum value.
+     * @param consider_remappings If true, component mappings will be honored
+     * if the original component is disabled.
      * @return The component associated with the value or a null pointer if no
      * such component exists.
      */
-    C* Lookup(EnumVal* val) const;
+    C* Lookup(EnumVal* val, bool consider_remappings = true) const;
+
+    /**
+     * Registers a mapping of a component to another one that will be honored
+     * by the `Lookup()` methods if (and only if) the original is currently
+     * disabled.
+     *
+     * @param old The original component tag.
+     * @param new_ The new component tag.
+     */
+    void AddComponentMapping(const zeek::Tag& old, const zeek::Tag& new_) {
+        if ( old != new_ ) {
+            component_mapping_by_src[old] = new_;
+            component_mapping_by_dst[new_] = old;
+        }
+    }
+
+    /**
+     * Returns true if a given component has a mapping to different one in place.
+     *
+     * @param tag The component tag to check.
+     */
+    auto HasComponentMapping(const zeek::Tag& tag) { return component_mapping_by_src.count(tag); }
+
+    /**
+     * Returns true if a given component is mapped to from a different one.
+     *
+     * @param tag The component tag to check.
+     */
+    auto ProvidesComponentMapping(const zeek::Tag& tag) { return component_mapping_by_dst.count(tag); }
```

Comment:
With the `auto` return type this would return a `size_t` from `count`, suggest to type explicitly.

```suggestion
    bool ProvidesComponentMapping(const zeek::Tag& tag) const { return component_mapping_by_dst.count(tag); }
```

Reply:
Alternatively, changing it to `!= 0` would be more explicit about what's happening here.